File: src/main/java/nl/enjarai/essentialsnt/Essentialsnt.java
package nl.enjarai.essentialsnt;

import com.mojang.brigadier.CommandDispatcher;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
import net.fabricmc.loader.api.FabricLoader;
import net.minecraft.server.MinecraftServer;
import nl.enjarai.essentialsnt.commands.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.File;
import java.util.Timer;
import java.util.TimerTask;

public class Essentialsnt implements ModInitializer {
    public static final Logger LOGGER = LogManager.getLogger("Essentialsn't");
    public static MinecraftServer SERVER;

    public static String VERSION = FabricLoader.getInstance().getModContainer("essentialsnt").get().getMetadata().getVersion().getFriendlyString();

    public static final File CONFIG_FILE = new File("config/essentialsnt.json");
    public static ConfigManager CONFIG = ConfigManager.loadConfigFile(CONFIG_FILE);

    @Override
    public void onInitialize() {
        ServerLifecycleEvents.SERVER_STARTING.register(Essentialsnt::onServerStarting);

        ManagementCommands.register();
        SpawnCommand.register();
        WarpCommand.register();
        WildCommand.register();
        MsgCommand.register();

        Timer timer = new Timer();
        timer.schedule(new TimerTask() {
            @Override
            public void run() {
                CONFIG.saveConfigFile(CONFIG_FILE);
            }
        }, 60000, 60000);

        LOGGER.info("Essentials bad lol, even i can do better");
    }

    private static void onServerStarting(MinecraftServer server) {
        SERVER = server;
    }
}

File: src/main/kotlin/nl/enjarai/essentialsnt/Essentialsnt.kt
package nl.enjarai.essentialsnt

import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents
import net.fabricmc.loader.api.FabricLoader
import net.minecraft.server.MinecraftServer
import nl.enjarai.essentialsnt.commands.*
import nl.enjarai.essentialsnt.commands.dayvote.DayVoteCommand
import nl.enjarai.essentialsnt.config.GeneralConfig
import nl.enjarai.essentialsnt.config.MessagesConfig
import nl.enjarai.essentialsnt.config.ModConfig
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.util.*

object Essentialsnt : ModInitializer {
    const val MODID = "essentialsnt"
    val LOGGER: Logger = LogManager.getLogger(MODID)
    val CONFIG_DIR = FabricLoader.getInstance().configDir.resolve(MODID)

    lateinit var GENERAL_CONFIG: GeneralConfig
    lateinit var MESSAGES_CONFIG: MessagesConfig

    lateinit var SERVER: MinecraftServer

    override fun onInitialize() {
        ServerLifecycleEvents.SERVER_STARTING.register { SERVER = it }

        CONFIG_DIR.toFile().mkdir()
        reloadConfig()

        ManagementCommands.register()
        SpawnCommands.register()
        WarpCommands.register()
        WildCommands.register()
        MessageCommands.register()
        DayVoteCommand.register()

        val timer = Timer()
        timer.schedule(object : TimerTask() {
            override fun run() {
                GENERAL_CONFIG.save()
                MESSAGES_CONFIG.save()
            }
        }, 60000, 60000)

        LOGGER.info("Essentials is bad lol, even i can do better")
    }

    fun reloadConfig() {
        GENERAL_CONFIG = ModConfig.loadConfigFile(GeneralConfig::class.java)
        MESSAGES_CONFIG = ModConfig.loadConfigFile(MessagesConfig::class.java)
    }
}